src/client/components/builder/logic/DateResult.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useEffect, useState } from 'react'
+import { isValidExpression } from '../../../core/evaluator'
+import { BiCalendar, BiChevronDown } from 'react-icons/bi'
+import {
+  VStack,
+  HStack,
+  Box,
+  Text,
+  Input,
+  Badge,
+  Button,
+  Menu,
+  MenuButton,
+  MenuItem,
+  MenuList,
+  NumberInput,
+  NumberInputField,
+} from '@chakra-ui/react'
+
+import { useCheckerContext } from '../../../contexts'
+import { createBuilderField, OperationFieldComponent } from '../BuilderField'
+import { BuilderActionEnum, ConfigArrayEnum } from '../../../../util/enums'
+import update from 'immutability-helper'
+
+interface DateState {
+  variableId: string
+  isAdd: boolean
+  numberOfIntervals: number
+}
+
+const EMPTY_STATE: DateState = {
+  variableId: '',
+  isAdd: true,
+  numberOfIntervals: 0,
+}
+
+// TODO: Make more rigorous to include other time units
+const fromExpression = (expression: string): DateState => {
+  const splitExpression = expression.split(' ')
```

Comment:
Opted to just split the expression string rather than parsing like in `ConditionalResult`